### PR TITLE
PWGHF: updated PID in candidateSelectorLcToK0sP

### PR DESCRIPTION
**Pull request by DanielSamitz**
- updated PID section to use the new `TrackSelectorPID::getStatusTrackPIDTpcAndTof` introduced in #2109
- split standard and Bayesian PID into separate process functions `processWithStandardPID` and `processWithBayesPID`. This way, one does not have to run the task `o2-analysis-pid-bayes` when not using Bayesian PID (until now it always needs to be included).
- removed debug messages

### CODE REVIEW

File: PWGHF/TableProducer/candidateSelectorLcToK0sP.cxx [hunk 1]
```diff
@@ -16,66 +16,54 @@
...
+  Configurable<double> nSigmaTpcMaxHighP{"nSigmaTpcMaxHighP", 0., "Max nSigma in TPC for bachelor at high p"};
+  Configurable<double> nSigmaTpcCombinedMaxHighP{"nSigmaTpcCombinedMaxHighP", 9999., "Max nSigma in TPC combined with TOF for bachelor at high p"};
+  // TOF
+  Configurable<double> nSigmaTofMaxLowP{"nSigmaTofMaxLowP", 0., "Max nSigma in TOF for bachelor at low p"};
+  Configurable<double> nSigmaTofCombinedMaxLowP{"nSigmaTofCombinedMaxLowP", 9999., "Max nSigma in TOF combined with TPC for bachelor at low p"};
+  Configurable<double> nSigmaTofMaxHighP{"nSigmaTofMaxHighP", 3.0, "Max nSigma in TOF for bachelor at high p"};
+  Configurable<double> nSigmaTofCombinedMaxHighP{"nSigmaTofCombinedMaxHighP", 3.0, "Max nSigma in TOF combined with TPC for bachelor at high p"};
+  // Bayesian
+  Configurable<double> probBayesMinLowP{"probBayesMinLowP", 0.8, "min. Bayes probability for bachelor at low p [%]"};
+  Configurable<double> probBayesMinHighP{"probBayesMinHighP", 0.8, "min. Bayes probability for bachelor at high p [%]"};
```
Comment (DanielSamitz) [inline PWGHF/TableProducer/candidateSelectorLcToK0sP.cxx:53]: All the PID related configurables for the two different momentum regions could also be put into a `Configurable<LabeledArray<double>> cutsPID{ ...` , analogous to the topological cuts. I have no strong opinion on that.

**Comment by DanielSamitz** [comment]
@lvermunt @xinyepeng

**Review by lvermunt — APPROVED** [review]

File: PWGHF/TableProducer/candidateSelectorLcToK0sP.cxx [hunk 2]
```diff
@@ -16,66 +16,54 @@
...
+  void init(InitContext&)
   {
-    // this is for now just a placeholder, in case we want to add extra checks
-    return true;
+    if (doprocessWithStandardPID == false && doprocessWithBayesPID == false) {
+      LOGF(fatal, "Neither processWithStandardPID nor processWithBayesPID enabled. Please choose one.");
+    }
+    if (doprocessWithStandardPID == true && doprocessWithBayesPID == true) {
+      LOGF(fatal, "Cannot enable processWithStandardPID and processWithBayesPID at the same time. Please choose one.");
+    }
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateSelectorLcToK0sP.cxx:66]:
```suggestion
    if (!doprocessWithStandardPID && !doprocessWithBayesPID) {
      LOGF(fatal, "Neither processWithStandardPID nor processWithBayesPID enabled. Please choose one.");
    }
    if (doprocessWithStandardPID && doprocessWithBayesPID) {
      LOGF(fatal, "Cannot enable processWithStandardPID and processWithBayesPID at the same time. Please choose one.");
    }
```
Use the booleans directly

File: PWGHF/TableProducer/candidateSelectorLcToK0sP.cxx [hunk 3]
```diff
@@ -143,101 +122,107 @@ struct HfCandidateSelectorLcToK0sP {
...
+        continue;
+      }
+
+      statusLc = 1;
+
+      hfSelLcToK0sPCandidate(statusLc);
+    }
+  }
+
+  PROCESS_SWITCH(HfCandidateSelectorLcToK0sP, processWithStandardPID, "Use standard PID for bachelor track", true);
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateSelectorLcToK0sP.cxx:194]:
```suggestion
  }
  PROCESS_SWITCH(HfCandidateSelectorLcToK0sP, processWithStandardPID, "Use standard PID for bachelor track", true);
```

File: PWGHF/TableProducer/candidateSelectorLcToK0sP.cxx [hunk 4]
```diff
@@ -143,101 +122,107 @@ struct HfCandidateSelectorLcToK0sP {
...
       }
 
       statusLc = 1;
-      MY_DEBUG_MSG(isLc && pidProton == 1, LOG(info) << "In selector: Lc ACCEPTED");
 
       hfSelLcToK0sPCandidate(statusLc);
     }
   }
+
+  PROCESS_SWITCH(HfCandidateSelectorLcToK0sP, processWithBayesPID, "Use Bayesian PID for bachelor track", false);
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateSelectorLcToK0sP.cxx:221]:
```suggestion
  }
  PROCESS_SWITCH(HfCandidateSelectorLcToK0sP, processWithBayesPID, "Use Bayesian PID for bachelor track", false);
```

**Review by vkucera — CHANGES_REQUESTED** [review]
Thanks @DanielSamitz , it looks good to me. Once you address the minor fixes, I'm happy to merge it.

**Review by vkucera — APPROVED** [review]

**Pull request merged**